pkg/api/v1/service/util.go

```diff
@@ -76,6 +76,19 @@ func RequestsOnlyLocalTraffic(service *v1.Service) bool {
 	return service.Spec.ExternalTrafficPolicy == v1.ServiceExternalTrafficPolicyTypeLocal
 }
 
+// RequestsOnlyLocalTrafficForInternal checks if service prefers Node Local
+// endpoints for internal traffic
+func RequestsOnlyLocalTrafficForInternal(service *v1.Service) bool {
+	if service.Spec.InternalTrafficPolicy == nil {
+		return false
+	}
+	switch *service.Spec.InternalTrafficPolicy {
+	case v1.ServiceInternalTrafficPolicyLocal:
+		return true
+	}
+	return false
+}
```

Comment:
Although this likely came from when there were more policies, this feels a bit overcomplicated now.

Reply:
This is to be aligned with externalTrafficPolicy: https://github.com/kubernetes/kubernetes/pull/96600/files/d0f3fd726b6a470c56d70e904db2407f9f345ae3#diff-514495aa2eda84bb5aed2784b8f2314e75b2f2861a369ef14c4a9885382f68eeR70

But I'm open to changes.

Reply:
updated